Density/app/src/main/java/org/cornelldti/density/density/facilitydetail/FacilityInfoPage.kt

```diff
@@ -396,117 +367,103 @@ class FacilityInfoPage : BaseActivity() {
-     * This function sets the operating hours text under the historical data chart
-     * @param day The day for which the operating hours is set
-     */
-    private fun setOperatingHoursText(day: String, date: Date) {
-        Log.d("SET", "OPERATING")
-        val hourTitle = FluxUtil.dayFullString(day)
-        todayHours.text = hourTitle
-        val todayDateFormat = SimpleDateFormat("MMMMM dd", Locale.US)
-        todayDate.text = todayDateFormat.format(date)
-        val todayDayDateFormat = SimpleDateFormat("EEE, MMM dd", Locale.US)
-        todayDayDate.text = todayDayDateFormat.format(date)
-        if (opHoursStrings.isEmpty()) facilityHours.text = getString(R.string.no_operating_hours)
-        else facilityHours.text = ""
-        for (operatingSegment in opHoursStrings) {
-            val allHours = facilityHours.text.toString() + operatingSegment + if (opHoursStrings.indexOf(operatingSegment) == opHoursStrings.size - 1) "" else "\n"
-            facilityHours.text = allHours
-        }
-    }
-
     override fun updateUI() {
         Log.d("updatedFPUI", "updating")
         fetchOperatingHours(date = FluxUtil.getDateObject(selectedDay!!)) // TODO FIX!! ON REFRESH!!
     }
 
-    private fun fetchHistoricalJSON(day: String, facilityId: String) {
-        api.fetchHistoricalJSON(
-                day = day,
-                facilityId = facilityId,
-                fetchHistoricalJSONOnResponse = { historicalDensities ->
-                    densities = historicalDensities
-                    setupBarChart()
-                }
-        )
-    }
-
     private fun fetchMenuJSON(day: String, facilityId: String) {
         api.fetchMenuJSON(
                 day = day,
                 facilityId = facilityId
         ) { menu ->
+            // set the current menu to the fetched menu
+            currentMenu = menu
+
+            // once loaded, hide the loader
             menuProgressBar.isGone = true
-            if (menu?.breakfastItems?.size == 0
+
+            val dayAvailableMenus = mutableListOf<String>()
+            val dayAvailableMenuTabs = mutableListOf<TabLayout.Tab>()
+
+            if (menu?.breakfastItems?.size != 0) {
+                dayAvailableMenuTabs.add(menuTabs.newTab().setText(R.string.breakfast))
+                dayAvailableMenus.add("breakfast")
+            }
+            if (menu?.brunchItems?.size != 0) {
+                dayAvailableMenuTabs.add(menuTabs.newTab().setText(R.string.brunch))
+                dayAvailableMenus.add("brunch")
+            }
+            if (menu?.lunchItems?.size != 0) {
+                dayAvailableMenuTabs.add(menuTabs.newTab().setText(R.string.lunch))
+                dayAvailableMenus.add("lunch")
+            }
+            if (menu?.dinnerItems?.size != 0) {
+                dayAvailableMenuTabs.add(menuTabs.newTab().setText(R.string.dinner))
+                dayAvailableMenus.add("dinner")
+            }
+
+            if (menu?.breakfastItems?.isEmpty() == true
```

Comment:
== true is redundant

Reply:
Take a look at some of the discussion above. I describe why it's necessary. https://github.com/cornell-dti/campus-density-android/pull/55#discussion_r499300111